### PR TITLE
Regression support

### PR DESCRIPTION
**Pull request by bobcheng15**
1. Update README to fix typo and description of the density input argument
2. Update the random matrix generation code, it now supports all datatype and uses density instead of nnz as input
3. Update code gen for rtl so it dumps out the shape of the gold, this is required by the future regression
4. Updated codegen to support identity function like X(i, j) = B(i, j)
5. Added input languages for the apps working on the future aha seed flow regression
6. Move activation specification to the language, can apply activation at either the ap, cp, or the cgra level
7. Added codegen for handling reading in empty matrix after rtl/comal computation
8. Change the no_preprocess to store_true action instead of passing "yes" or "no" as a string

### CODE REVIEW

File: pre_process.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def process(tensor_type, input_path, output_dir_path, tensor_size, schedule_dict
         if dtype == "int":
             for idx, val in np.ndenumerate(tensor):
                 if val != 0:
-                    tensor[idx] = np.random.randint(1, 100)
```
Comment (sgauthamr2001) [inline pre_process.py:227]: Probs, np.random.randint(-100, 100)
and if tensor[idx] == 0, set to 1

File: src/data_parser.cpp [hunk 2]
```diff
@@ -207,7 +212,7 @@ int subtile_paths_printer(const std::vector<std::string> &subtile_paths,
 			if (i != std::string::npos) {
 				subtile_path.erase(i, path_prefix.length());
 			}
-
+			std::cout << subtile_path << std::endl;
```
Comment (sgauthamr2001) [inline src/data_parser.cpp:215]: Debug flag??

**Review by sgauthamr2001 — COMMENTED** [review]
Minor code changes

File: src/data_parser.cpp (hunk 2 above)
Comment (bobcheng15) [reply src/data_parser.cpp]: removed

File: pre_process.py (hunk 1 above)
Comment (bobcheng15) [reply pre_process.py:227]: fixed

**Comment by bobcheng15** [comment]
@sgauthamr2001 I have patched the two code locations, lmk if there's anything wrong

**Pull request merged**